src/lib/alloc.c

```diff
@@ -202,7 +202,7 @@ static void *align_ptr(struct mm_heap *heap, uint32_t alignment,
 
 /* allocate single block */
 static void *alloc_block(struct mm_heap *heap, int level,
-			 uint32_t caps, uint32_t alignment)
+			 uint32_t alignment)
```

Comment:
ehm, they should be using this. Not all memory is equal. This fix should probably be a new PR.

Reply:
ok, let's drop / postpone number 3 then, because they're interdependent